crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -17,23 +17,31 @@ use std::time::Duration;
 
 use futures_util::stream::{FuturesUnordered, Stream, StreamExt, once};
 use smallvec::SmallVec;
-use tracing::debug;
+use tracing::{debug, error};
 
 use crate::config::{
     NameServerConfig, OpportunisticEncryption, ResolverOpts, ServerOrderingStrategy,
     SharedNameServerTransportState,
 };
 use crate::name_server::connection_provider::{ConnectionProvider, TlsConfig};
 use crate::name_server::name_server::{ConnectionPolicy, NameServer};
-use crate::proto::op::{DnsRequest, DnsResponse, ResponseCode};
-use crate::proto::runtime::{RuntimeProvider, Time};
-use crate::proto::xfer::DnsHandle;
-use crate::proto::{DnsError, NoRecords, ProtoError, ProtoErrorKind};
+use crate::proto::{
+    DnsError, NoRecords, ProtoError, ProtoErrorKind,
+    op::{DnsRequest, DnsResponse, ResponseCode},
+    rr::{
+        RData, Record,
+        rdata::{A, AAAA},
+    },
+    runtime::{RuntimeProvider, Time},
+    util::accesscontrol::AccessControlSet,
+    xfer::DnsHandle,
+};
 
 /// Abstract interface for mocking purpose
 #[derive(Clone)]
 pub struct NameServerPool<P: ConnectionProvider> {
     state: Arc<PoolState<P>>,
+    answer_address_filter: Option<Arc<AccessControlSet>>,
```

Comment:
Suggest we stick this in the `PoolContext` instead? I don't think it needs to change over time.